### PR TITLE
[PWA-2149] Fixing submenu column error

### PR DESCRIPTION
**Pull request by JustinConabreeAbsolunet**
## Description

Fixing error `Cannot read properties of undefined (reading 'length')` when subcategory doesn't have children
Adding fix from [PR #3430](https://github.com/magento/pwa-studio/pull/3430)

## Related Issue

Closes [PWA-2149](https://jira.corp.magento.com/browse/PWA-2149)

#### Test scenario(s) for direct fix/feature

Load homepage with a backend that has a similar structure to the following
```
Level 1 Category 1
-- Level 2 Category 1
---- Level 3 Category 1
------ Level 4 Category 1
------ Level 4 Category 2
---- Level 3 Category 2
---- Level 3 Category 1
-- Level 2 Category 2
---- Level 3 Category 1
```
Previously it would fail at `---- Level 3 Category 2` since there were no children. Base Luma sample data has a structure that should cause issue.

## Breaking Changes (if any)
None

## Checklist

- ~I have added tests to cover my changes, if necessary.~
- ~I have added translations for new strings, if necessary.~
- ~I have updated the documentation accordingly, if necessary.~

### CODE REVIEW

**Review by larsroettig — APPROVED** [review]

**Comment by revanth0212** [comment]
@larsroettig how did you test this change? Do you happen to have a backend that supports the verification steps @JustinConabreeAbsolunet provided?

File: packages/venia-ui/lib/components/MegaMenu/submenuColumn.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ const SubmenuColumn = props => {
     let children = null;
 
     if (category.children.length) {
-        const childrenItems = category.children.map((category, index) => {
-            const { url_path, isActive, name } = category;
+        const childrenItems = category.children.map((subCategory, index) => {
```
Comment (revanth0212) [inline packages/venia-ui/lib/components/MegaMenu/submenuColumn.js:26]: Isn't this a variable name change? How did this fix the issue?

Comment (JustinConabreeAbsolunet) [reply packages/venia-ui/lib/components/MegaMenu/submenuColumn.js:26]: Line 25 checks for category.children.length, but then line 26 was changing the category variable. A couple lines later we would evaluate category.children.length, but it was undefined on the sub category. This fix is just undoing the variable overwritting that was happening

**Comment by revanth0212** [comment]
Setup the categories like the verification steps mentioned and able to reproduce the issue

image

Fix:

image

**Comment by revanth0212** [comment]
Hey @JustinConabreeAbsolunet. Some cypress tests are failing:

image

PB Row and Multiple wishlist tests are failing on develop as well so you can ignore them. Others might be something related to your PR changes.

To run the tests: `yarn test:headless https://pr-3427.pwa-venia.com/`
To update the failing snapshots: `yarn test:headless https://pr-3427.pwa-venia.com/ true`

If you update the snapshots please only make sure to push the ones that are related to your change.

**Comment by shikhamis11** [comment]
@JustinConabreeAbsolunet  After applying this fix, the original issue is gone but I am getting `Unhandled InvariantError43U Invariant Violation: Invariant Violation: 33 (see https://github.com/apollographql/invariant-packages)
` error now
WhatsApp Image 2021-09-22 at 12 35 18 PM

I am able to reproduce this error with Magento 2.4.3  backend with Sample data installed.

**Comment by JustinConabreeAbsolunet** [comment]
Hey @shikhamis11 

Do you have any more details about how to reproduce? I'm running the branch with a vanilla 2.4.3 backend with sample data and I don't have the issue. This change is to a presentation component so it shouldn't have any affect on Apollo.

Some questions that might help:
- Did you apply the fix to a different branch? Or checking out this branch has the issue?
- Do you have the issue when you checkout the `develop` branch?
- Have you tried clearing the application storage?
- How are you booting the application? (`yarn watch:all` vs `yarn build` and `yarn stage:venia`)
- Try a `yarn install` before your watch/build
- Which page are you trying to load? (or is it every page?)

Thanks!

**Comment by shikhamis11** [comment]
Hi @JustinConabreeAbsolunet 
I just clone the PWA studio repo and did the installation by following steps from [devdocs](https://magento.github.io/pwa-studio/venia-pwa-concept/setup/) 

after installation, I changed the `MAGENTO_BACKEND_URL` with my instance(Magento 2.4.3 backend with Sample data installed)
I also changed `MAGENTO_BACKEND_EDITION` to CE and applied the fix . 

then just run 
`yarn run build` && `yarn run stage:venia`

I did run `yarn install` while installation 
I have added a short video of error please check 

https://user-images.githubusercontent.com/25526037/134357432-bd4e7182-5f9e-407b-9654-f5632b7a2c77.mp4

**Comment by JustinConabreeAbsolunet** [comment]
Hi @shikhamis11 ,

Can you please try applying the changes found in this PR? https://github.com/magento/pwa-studio/pull/3430/files
I spoke with the team and it seems like the issue you're having has been resolved in a separate branch that is in progress (unrelated to this PR).

**Comment by shikhamis11** [comment]
Hi @JustinConabreeAbsolunet, 
Thanks for the update. Yes after applying the fix from PR https://github.com/magento/pwa-studio/pull/3430/files 
It is working fine. There was no issue with the default backend URL so I thought the error is after the fix. 
Thanks again for a quick update.

**Comment by JustinConabreeAbsolunet** [comment]
@shikhamis11 Thanks for the callout! We actually added the fix to this PR since it's critical.

**Comment by revanth0212** [comment]
Merging even though `pwa-pw-test` is failing because it's a bug outside the scope of this PR. It is being handled in a different PR.

**Pull request merged**